internal/services/docker/docker_api.go

```diff
@@ -103,20 +103,28 @@ func (d *API) CreateLanguageAnalysisContainer(data *docker.AnalysisData) (contai
 	return d.logStatusAndExecuteCRDContainer(data.GetCustomOrDefaultImage(), d.replaceCMDAnalysisID(data.CMD))
 }
 
+// PullImage check if an image already exists on cache, if its not, pull from registry.
+//
+// nolint:funlen
 func (d *API) PullImage(imageWithTagAndRegistry string) error {
 	if d.config.DisableDocker {
 		return nil
 	}
 
-	if imageNotExist, err := d.checkImageNotExists(imageWithTagAndRegistry); err != nil || !imageNotExist {
+	imageNotExist, err := d.checkIfImageNotExists(imageWithTagAndRegistry)
+	if err != nil {
 		logger.LogError(fmt.Sprintf("%s -> %s",
 			messages.MsgErrorFailedToPullImage, imageWithTagAndRegistry), err)
 		return err
+	} else if imageNotExist {
+		logger.LogDebugWithLevel(fmt.Sprintf(messages.MsgDebugDockerImageDoesNotExists, imageWithTagAndRegistry))
+		if err := d.downloadImage(imageWithTagAndRegistry); err != nil {
```

Comment:
`LogError` will log only when error different than nil, unnecessary if statement

Reply:
I think that using if make more clear, since in some places we use and other places we not.

Reply:
Still think it's unnecessary, maybe change the log function name to be more clear about the validation, something like `LogErrorIfExists`.

In the current way we are repeating unnecessary code.
```
if err != nil {
  if err != nil {
    ...
  }
}
```

Reply:
I'ill remove this validation but I don't think it's the logger's role to check for errors.

Reply:
It might be better to make it clear that the function already do the validation and avoid performing it again.